test.lua

```diff
@@ -16,6 +17,54 @@ strict.on()
 --                          local support functions                          --
 -- ========================================================================= --
 
+-- Strip pcall()'s true or re-raise catched error.
+local function wait_cond_finish(res)
+    -- res: 1: pcall_ok
+    --      2: pcall_error / cond_function_res_1
+    --      3: cond_function_res_2
+    --      4: cond_function_res_3
+    --      ...
+    --
+    -- cond_function_res_1 is the 'ok' / 'not ok' indicator for
+    -- wait_cond(). But we return it: this is deliberate choice.
+    -- This way a user may understand, whether we finally hit
+    -- timeout or not.
+    if not res[1] then
+        error(res[2])
+    end
+    return select(2, unpack(res))
```

Comment:
unpack for {1, nil, 3} returns only 1. Should be replaced to unpack(res, 1, table.maxn(res)).

Reply:
Good point. Fixed.